astropy/coordinates/angles.py

```diff
@@ -644,12 +634,20 @@ def _wrap_internal(self):
         # Convert the wrap angle and 360 degrees to the native unit of
         # this Angle, then do all the math on raw Numpy arrays rather
         # than Quantity objects for speed.
+        # TODO: This could probably be sped up as well.  u.degree.to(self.unit)
+        # should be faster if self.unit *is* u.degree
         a360 = u.degree.to(self.unit, 360.0)
-        wrap_angle = self.wrap_angle.to(self.unit).value
+
+        # TODO: Quantity.to is currently slow when the quantity is already in
```

Comment:
Sure.  And I agree, as code it sucks, but I think it's an area where practicality beats purity.